src/Geta.404Handler/Core/CustomRedirects/CustomRedirectCollection.cs

```diff
@@ -109,7 +110,7 @@ private CustomRedirect FindInternal(string url)
             // url to the <new> url.
             foreach (var redirectPair in redirectsZA)
             {
-                var oldUrl = redirectPair.Key;
+                var oldUrl = Url.Encode(redirectPair.Key);
```

Comment:
I see the null check below that checks if `oldUrl` is not null. Does the `Url.Encode` accept null as a parameter?